packages/ipfs/src/core/components/id.js

```diff
@@ -9,10 +9,12 @@ module.exports = ({ peerId, libp2p }) => {
   return withTimeoutOption(async function id () { // eslint-disable-line require-await
     const id = peerId.toB58String()
     let addresses = []
+    let protocols = []
 
     if (libp2p) {
       // only available while the node is running
       addresses = libp2p.transportManager.getAddrs()
+      protocols = libp2p.peerStore.protoBook.get(peerId) || []
```

Comment:
Self protocols aren't currently stored in the peer store, so they need to be retrieved from the upgrader for now.
```suggestion
      protocols = Array.from(libp2p.upgrader.protocols.keys())
```

Reply:
Done! Thanks @jacobheun